### PR TITLE
Refactor/116 RegistryInfoContent is converted to functional component

### PR DESCRIPTION
**Pull request by agar23**
**Type:** Refactor

The following has been addressed in the PR:

*  There is a related issue? #116 
*  Unit or Functional tests are included in the PR: Yes

**Description:**

Refactor RegistryInfoContent compoenent to functional

### CODE REVIEW

**Comment by agar23** [comment]
@priscilawebdev @juanpicado  please review.

File: src/components/Header/HeaderInfoDialog.tsx [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import React from 'react';
 
 import RegistryInfoDialog from '../RegistryInfoDialog';
-import RegistryInfoContent from '../RegistryInfoContent';
+import { RegistryInfoContent } from '../RegistryInfoContent';
```
Comment (priscilawebdev) [inline src/components/Header/HeaderInfoDialog.tsx]: Hmm... I would rather leave it as a default import as it it the main code of `RegistryInfoContent`

File: src/components/PackageList/PackageList.tsx [hunk 2]
```diff
@@ -1,4 +1,4 @@
-import * as React from 'react';
+import React, { Fragment, ReactElement } from 'react';
```
Comment (priscilawebdev) [inline src/components/PackageList/PackageList.tsx]: Maybe we should discuss about some standards, but I am trying to keep the imports together, like you did here : 
image

File: src/components/PackageList/PackageList.tsx [hunk 3]
```diff
@@ -13,16 +13,16 @@ interface Props {
 }
 
 export const PackageList: React.FC<Props> = props => {
-  const renderPackages: () => React.ReactElement<HTMLElement>[] = () => {
+  const renderPackages: () => ReactElement<HTMLElement>[] = () => {
```
Comment (priscilawebdev) [inline src/components/PackageList/PackageList.tsx]: Hm...I don't know if I agree with this type... maybe React.ReactNode?

File: src/components/RegistryInfoContent/RegistryInfoContent.test.tsx [hunk 4]
```diff
@@ -1,23 +1,24 @@
...
+    const unorderedListOfTodos = getByTestId('tabs-el');
+    expect(unorderedListOfTodos.children.length).toBe(1);
   });
 
-  test('should render the component in default state with yarn tab', () => {
-    const wrapper = mount(<RegistryInfoContent registryUrl="https://registry.verdaccio.org" scope="@" />);
-    wrapper.setState({ tabPosition: 2 });
-    expect(wrapper.html()).toMatchSnapshot();
+  test('should load the appropiate tab content when the tab is clicked', () => {
+    const { container, getByTestId } = render(<RegistryInfoContent registryUrl={'http://localhost:4872'} scope={''} />);
```
Comment (priscilawebdev) [inline src/components/RegistryInfoContent/RegistryInfoContent.test.tsx]: I would keep the parameters in some object, because it's easy so do maintenance.

File: src/components/RegistryInfoContent/RegistryInfoContent.test.tsx [hunk 5]
```diff
@@ -1,23 +1,24 @@
...
+    expect(container.textContent).not.toContain('pnpm adduser --registry http://localhost:4872');
```
Comment (priscilawebdev) [inline src/components/RegistryInfoContent/RegistryInfoContent.test.tsx]: if you use an object as I mentioned above you could reuse the registryURL here :-)

File: src/components/RegistryInfoContent/RegistryInfoContent.tsx [hunk 6]
```diff
@@ -1,4 +1,4 @@
-import React, { Component } from 'react';
+import React, { useState } from 'react';
```
Comment (priscilawebdev) [inline src/components/RegistryInfoContent/RegistryInfoContent.tsx]: 👏

File: src/components/RegistryInfoContent/RegistryInfoContent.tsx [hunk 7]
```diff
@@ -11,86 +11,79 @@ import { default as Typography } from '../../muiComponents/Heading';
...
-    tabPosition: 0,
-  };
-
-  public render(): JSX.Element {
-    return <div>{this.renderTabs()}</div>;
-  }
-
-  private handleChange = (event: React.ChangeEvent<{}>, tabPosition: number) => {
+export const RegistryInfoContent: React.FC<Props> = props => {
+  const [tabPosition, setTabPosition] = useState<State['tabPosition']>(0);
```
Comment (priscilawebdev) [inline src/components/RegistryInfoContent/RegistryInfoContent.tsx]:
```suggestion
  const [tabPosition, setTabPosition] = useState(0);
```

Here you don't need to set a type, it will be automatically of type number, because of it's value, 0.

File: src/components/RegistryInfoContent/RegistryInfoContent.tsx [hunk 8]
```diff
@@ -11,86 +11,79 @@ import { default as Typography } from '../../muiComponents/Heading';
...
-        {tabPosition === 1 && <TabContainer>{this.renderPNpmTab(scope, registryUrl)}</TabContainer>}
-        {tabPosition === 2 && <TabContainer>{this.renderYarnTab(scope, registryUrl)}</TabContainer>}
-      </React.Fragment>
-    );
-  }
-
-  private renderNpmTab(scope: string, registryUrl: string): JSX.Element {
+  const renderNpmTab = (scope: string, registryUrl: string): JSX.Element => {
     return (
       <React.Fragment>
```
Comment (priscilawebdev) [inline src/components/RegistryInfoContent/RegistryInfoContent.tsx]:
```suggestion
      <>
```

File: src/components/RegistryInfoContent/RegistryInfoContent.tsx [hunk 9]
```diff
@@ -11,86 +11,79 @@ import { default as Typography } from '../../muiComponents/Heading';
...
         <CopyToClipBoard text={getCLISetConfigRegistry(`${NODE_MANAGER.npm} set`, scope, registryUrl)} />
         <CopyToClipBoard text={getCLISetRegistry(`${NODE_MANAGER.npm} adduser`, registryUrl)} />
         <CopyToClipBoard text={getCLIChangePassword(NODE_MANAGER.npm, registryUrl)} />
       </React.Fragment>
```
Comment (priscilawebdev) [inline src/components/RegistryInfoContent/RegistryInfoContent.tsx]:
```suggestion
      </>
```

File: src/components/RegistryInfoContent/RegistryInfoContent.tsx [hunk 10]
```diff
@@ -11,86 +11,79 @@ import { default as Typography } from '../../muiComponents/Heading';
...
         <CopyToClipBoard text={getCLIChangePassword(NODE_MANAGER.pnpm, registryUrl)} />
       </React.Fragment>
     );
-  }
+  };
 
-  private renderYarnTab(scope: string, registryUrl: string): JSX.Element {
+  const renderYarnTab = (scope: string, registryUrl: string): JSX.Element => {
     return (
       <React.Fragment>
```
Comment (priscilawebdev) [inline src/components/RegistryInfoContent/RegistryInfoContent.tsx]:
```suggestion
      <>
```

File: src/components/RegistryInfoContent/RegistryInfoContent.tsx [hunk 11]
```diff
@@ -11,86 +11,79 @@ import { default as Typography } from '../../muiComponents/Heading';
...
         <CopyToClipBoard text={getCLISetConfigRegistry(`${NODE_MANAGER.yarn} config set`, scope, registryUrl)} />
       </React.Fragment>
```
Comment (priscilawebdev) [inline src/components/RegistryInfoContent/RegistryInfoContent.tsx]:
```suggestion
      </>
```

File: src/components/RegistryInfoContent/RegistryInfoContent.tsx [hunk 12]
```diff
@@ -11,86 +11,79 @@ import { default as Typography } from '../../muiComponents/Heading';
...
         <CopyToClipBoard text={getCLIChangePassword(NODE_MANAGER.npm, registryUrl)} />
       </React.Fragment>
     );
-  }
+  };
 
-  private renderPNpmTab(scope: string, registryUrl: string): JSX.Element {
+  const renderPNpmTab = (scope: string, registryUrl: string): JSX.Element => {
     return (
       <React.Fragment>
```
Comment (priscilawebdev) [inline src/components/RegistryInfoContent/RegistryInfoContent.tsx]:
```suggestion
      <>
```

File: src/components/RegistryInfoContent/RegistryInfoContent.tsx [hunk 13]
```diff
@@ -11,86 +11,79 @@ import { default as Typography } from '../../muiComponents/Heading';
...
         <CopyToClipBoard text={getCLISetConfigRegistry(`${NODE_MANAGER.npm} set`, scope, registryUrl)} />
         <CopyToClipBoard text={getCLISetRegistry(`${NODE_MANAGER.npm} adduser`, registryUrl)} />
         <CopyToClipBoard text={getCLIChangePassword(NODE_MANAGER.npm, registryUrl)} />
       </React.Fragment>
     );
-  }
+  };
 
-  private renderPNpmTab(scope: string, registryUrl: string): JSX.Element {
+  const renderPNpmTab = (scope: string, registryUrl: string): JSX.Element => {
```
Comment (priscilawebdev) [inline src/components/RegistryInfoContent/RegistryInfoContent.tsx]:
```suggestion
  const renderPnpmTab = (scope: string, registryUrl: string): JSX.Element => {
```

File: src/components/RegistryInfoContent/RegistryInfoContent.tsx [hunk 14]
```diff
@@ -11,86 +11,79 @@ import { default as Typography } from '../../muiComponents/Heading';
...
     return (
       <React.Fragment>
         <CopyToClipBoard text={getCLISetConfigRegistry(`${NODE_MANAGER.pnpm} set`, scope, registryUrl)} />
         <CopyToClipBoard text={getCLISetRegistry(`${NODE_MANAGER.pnpm} adduser`, registryUrl)} />
         <CopyToClipBoard text={getCLIChangePassword(NODE_MANAGER.pnpm, registryUrl)} />
       </React.Fragment>
```
Comment (priscilawebdev) [inline src/components/RegistryInfoContent/RegistryInfoContent.tsx]:
```suggestion
      </>
```

File: src/components/RegistryInfoContent/index.ts [hunk 15]
```diff
@@ -1 +1 @@
-export { default } from './RegistryInfoContent';
```
Comment (priscilawebdev) [inline src/components/RegistryInfoContent/index.ts]: please try to keep the standard 😉

File: src/components/RegistryInfoContent/RegistryInfoContent.tsx [hunk 16]
```diff
@@ -11,86 +11,79 @@ import { default as Typography } from '../../muiComponents/Heading';
...
     );
-  }
-}
+  };
+
+  const renderTabs = (): JSX.Element => {
+    const { scope, registryUrl } = props;
+
+    return (
+      <React.Fragment>
```
Comment (priscilawebdev) [inline src/components/RegistryInfoContent/RegistryInfoContent.tsx]:
```suggestion
      <>
```

File: src/components/RegistryInfoContent/RegistryInfoContent.tsx [hunk 17]
```diff
@@ -11,86 +11,79 @@ import { default as Typography } from '../../muiComponents/Heading';
...
+          value={tabPosition}
+          variant="fullWidth">
+          <Tab data-testid={'npm-tab'} label={NODE_MANAGER.npm} />
+          <Tab data-testid={'pnpm-tab'} label={NODE_MANAGER.pnpm} />
+          <Tab data-testid={'yarn-tab'} label={NODE_MANAGER.yarn} />
+        </Tabs>
+        {tabPosition === 0 && <TabContainer>{renderNpmTab(scope, registryUrl)}</TabContainer>}
+        {tabPosition === 1 && <TabContainer>{renderPNpmTab(scope, registryUrl)}</TabContainer>}
+        {tabPosition === 2 && <TabContainer>{renderYarnTab(scope, registryUrl)}</TabContainer>}
+      </React.Fragment>
```
Comment (priscilawebdev) [inline src/components/RegistryInfoContent/RegistryInfoContent.tsx]:
```suggestion
      </>
```

File: src/components/PackageList/PackageList.tsx [hunk 18]
```diff
@@ -13,16 +13,16 @@ interface Props {
 }
 
 export const PackageList: React.FC<Props> = props => {
```
Comment (priscilawebdev) [inline src/components/PackageList/PackageList.tsx]: Please destruct the props here, then you don't need to to like this: 
image

**Comment by priscilawebdev** [comment]
@agar23 thank you very much for your contributing 🙂👏I would apply just a few minor changes ... please check my comments :-)

File: src/components/RegistryInfoContent/RegistryInfoContent.tsx (hunk 7 above)
Comment (tmkn) [reply src/components/RegistryInfoContent/RegistryInfoContent.tsx]: Nothing wrong with being explicit.
Setting a type is useful so that you can prevent a typo like this:
```typescript
const [tabPosition, setTabPosition] = useState('0'); //ok...
```
Without explicitly setting a type the compiler is fine with whatever you initialize the state with.
Now the type is `string` and any error because of type mismatch will show up where you consume the state, whilst if you would have been explicit:
```typescript
const [tabPosition, setTabPosition] = useState<number>('0'); //Argument of type '"0"' is not assignable to parameter of type 'number | (() => number)'
```
You would get an error right on this line, where the actual error happenend, since you **intended** the state to be of type `number` but (accidentally) passed in a `string` instead. :wink:

File: src/components/Header/HeaderInfoDialog.tsx (hunk 1 above)
Comment (agar23) [reply src/components/Header/HeaderInfoDialog.tsx]: generally speaking, using default imports/exports can lead to issues. Is there a reason why we prefer to use them vs named imports/exports?

https://basarat.gitbooks.io/typescript/docs/tips/defaultIsBad.html
https://stackoverflow.com/a/33307487

File: src/components/PackageList/PackageList.tsx (hunk 2 above)
Comment (agar23) [reply src/components/PackageList/PackageList.tsx]: yes! is there a weekly/monthly call or is discord the best forum for discussions?

File: src/components/RegistryInfoContent/index.ts (hunk 15 above)
Comment (agar23) [reply src/components/RegistryInfoContent/index.ts]: same as https://github.com/verdaccio/ui/pull/229#discussion_r339846597 wondering why we want to use default exports?

File: src/components/PackageList/PackageList.tsx (hunk 18 above)
Comment (agar23) [reply src/components/PackageList/PackageList.tsx]: we are not destructuring the props for this component.

**Comment by agar23** [comment]
> @agar23 thank you very much for your contribution 🙂👏I would apply just a few minor changes ... please check my comments :-)

np, I addressed most of the comments. but had a couple of questions/comments I left.

File: src/components/RegistryInfoContent/RegistryInfoContent.tsx (hunk 13 above)
Comment (agar23) [reply src/components/RegistryInfoContent/RegistryInfoContent.tsx]: done

File: src/components/RegistryInfoContent/RegistryInfoContent.test.tsx (hunk 5 above)
Comment (agar23) [reply src/components/RegistryInfoContent/RegistryInfoContent.test.tsx]: done

File: src/components/RegistryInfoContent/RegistryInfoContent.tsx (hunk 6 above)
Comment (agar23) [reply src/components/RegistryInfoContent/RegistryInfoContent.tsx]: 😄

File: src/components/RegistryInfoContent/RegistryInfoContent.tsx (hunk 7 above)
Comment (agar23) [reply src/components/RegistryInfoContent/RegistryInfoContent.tsx]: I agree although typescript is smart enough to infer the type here we should declare it. if not then we should remove our state interface.

File: src/components/RegistryInfoContent/RegistryInfoContent.tsx (hunk 8 above)
Comment (agar23) [reply src/components/RegistryInfoContent/RegistryInfoContent.tsx]: done

File: src/components/RegistryInfoContent/RegistryInfoContent.tsx (hunk 9 above)
Comment (agar23) [reply src/components/RegistryInfoContent/RegistryInfoContent.tsx]: done

File: src/components/RegistryInfoContent/RegistryInfoContent.tsx (hunk 11 above)
Comment (agar23) [reply src/components/RegistryInfoContent/RegistryInfoContent.tsx]: done

File: src/components/RegistryInfoContent/RegistryInfoContent.tsx (hunk 14 above)
Comment (agar23) [reply src/components/RegistryInfoContent/RegistryInfoContent.tsx]: done

File: src/components/RegistryInfoContent/RegistryInfoContent.tsx (hunk 16 above)
Comment (agar23) [reply src/components/RegistryInfoContent/RegistryInfoContent.tsx]: done

File: src/components/RegistryInfoContent/RegistryInfoContent.tsx (hunk 17 above)
Comment (agar23) [reply src/components/RegistryInfoContent/RegistryInfoContent.tsx]: done

File: src/components/RegistryInfoContent/RegistryInfoContent.tsx (hunk 12 above)
Comment (agar23) [reply src/components/RegistryInfoContent/RegistryInfoContent.tsx]: done

File: src/components/RegistryInfoContent/RegistryInfoContent.tsx (hunk 10 above)
Comment (agar23) [reply src/components/RegistryInfoContent/RegistryInfoContent.tsx]: done

File: src/components/RegistryInfoContent/RegistryInfoContent.test.tsx (hunk 4 above)
Comment (agar23) [reply src/components/RegistryInfoContent/RegistryInfoContent.test.tsx]: ohhh yes! done

File: src/components/PackageList/PackageList.tsx (hunk 3 above)
Comment (agar23) [reply src/components/PackageList/PackageList.tsx]: done

File: src/components/RegistryInfoContent/RegistryInfoContent.tsx (hunk 17 above)
Comment (agar23) [reply src/components/RegistryInfoContent/RegistryInfoContent.tsx]: done

File: src/components/PackageList/PackageList.tsx (hunk 18 above)
Comment (priscilawebdev) [reply src/components/PackageList/PackageList.tsx]: Thank you!! :-)

Comment (priscilawebdev) [reply src/components/PackageList/PackageList.tsx]: I would still destruct the props like this: `const PackageList: React.FC<Props> =({ packages }) => `

File: src/components/PackageList/PackageList.tsx [hunk 19]
```diff
@@ -13,21 +13,22 @@ interface Props {
...
-  const renderPackages: () => React.ReactElement<HTMLElement>[] = () => {
-    return props.packages.map((pkg, i) => {
+  const { packages } = props;
+  const renderPackages: () => ReactNode[] = () => {
+    return packages.map((pkg, i) => {
```
Comment (priscilawebdev) [inline src/components/PackageList/PackageList.tsx]: This is only a suggestion, but I would avoid creating another constant by doing this:
```suggestion
    return packages.map(({ name, version, description, time, keywords, dist, homepage, bugs, author }, i) => {
```

File: src/components/RegistryInfoContent/RegistryInfoContent.tsx (hunk 7 above)
Comment (priscilawebdev) [reply src/components/RegistryInfoContent/RegistryInfoContent.tsx]: Hmmm I find it unnecessary, because we are not really setting this state so dynamically.... and even if we were passing a prop, it would be checked though of the interface or if we pass for example a string in the setTabPosition we will get a type error... so IMO is fine if you want to cast the state, but  it is very verbose and unnecessary.

File: src/components/Header/HeaderInfoDialog.tsx (hunk 1 above)
Comment (priscilawebdev) [reply src/components/Header/HeaderInfoDialog.tsx]: @agar23 thank you for share these info with us... I believe that it's something we can discuss for the upcoming verdaccio version.. I  will really read more about it, but please could you leave it as default for now? Then if we decide to do as you suggested,  we can add this change in all components 😉

File: src/components/PackageList/PackageList.tsx (hunk 2 above)
Comment (priscilawebdev) [reply src/components/PackageList/PackageList.tsx]: yes you can always contact us on verdaccio discord chat 😃This week I'm a little absent, but surely the other members will be online to help you

Comment (agar23) [reply src/components/PackageList/PackageList.tsx]: sounds good 👍

File: src/components/Header/HeaderInfoDialog.tsx (hunk 1 above)
Comment (agar23) [reply src/components/Header/HeaderInfoDialog.tsx]: sounds good 😄

File: src/components/RegistryInfoContent/RegistryInfoContent.tsx (hunk 7 above)
Comment (agar23) [reply src/components/RegistryInfoContent/RegistryInfoContent.tsx]: understood

File: src/components/PackageList/PackageList.tsx (hunk 18 above)
Comment (agar23) [reply src/components/PackageList/PackageList.tsx]: ahhh I see what you mean! done.

File: src/components/PackageList/PackageList.tsx (hunk 19 above)
Comment (agar23) [reply src/components/PackageList/PackageList.tsx]: ohh yes! done

File: src/components/PackageList/PackageList.tsx [hunk 20]
```diff
@@ -12,22 +12,23 @@ interface Props {
...
-export const PackageList: React.FC<Props> = props => {
-  const renderPackages: () => React.ReactElement<HTMLElement>[] = () => {
-    return props.packages.map((pkg, i) => {
-      const { name, version, description, time, keywords, dist, homepage, bugs, author } = pkg;
+export const PackageList: React.FC<Props> = ({ packages }) => {
+  const renderPackages: () => ReactNode[] = () => {
+    return packages.map(({ name, version, description, time, keywords, dist, homepage, bugs, author, license }, i) => {
       // TODO: move format license to API side.
-      const license = formatLicense(pkg.license);
+      const _license = formatLicense(license);
```
Comment (priscilawebdev) [inline src/components/PackageList/PackageList.tsx]: Please why did you add `_` ?

File: src/components/RegistryInfoContent/RegistryInfoContent.tsx [hunk 21]
```diff
@@ -11,86 +11,77 @@ import { default as Typography } from '../../muiComponents/Heading';
...
-          min-height: 170;
-        `}
-        component="div">
-        {children}
-      </Typography>
-    </CommandContainer>
-  );
-}
+const RegistryInfoContent: React.FC<Props> = props => {
+  const [tabPosition, setTabPosition] = useState<State['tabPosition']>(0);
```
Comment (priscilawebdev) [inline src/components/RegistryInfoContent/RegistryInfoContent.tsx]:
```suggestion
  const [tabPosition, setTabPosition] = useState(0);
```

**Comment by priscilawebdev** [comment]
@agar23 almost there :-)

**Comment by juanpicado** [comment]
@agar23 great job, there are one conflict and seems linting is failing :) looking forward to merge it soon :)

File: src/components/PackageList/PackageList.tsx (hunk 20 above)
Comment (agar23) [reply src/components/PackageList/PackageList.tsx]: we cant name this variable `license` the props already have a prop named `license` I presume this why the component was not destructuring the props before. this way we could call `pkg.license`

File: src/components/RegistryInfoContent/RegistryInfoContent.tsx (hunk 21 above)
Comment (agar23) [reply src/components/RegistryInfoContent/RegistryInfoContent.tsx]: as discussed I think it should be ok to be explicit here
